src/object.d

```diff
@@ -1947,7 +1947,12 @@ class TypeInfo_Struct : TypeInfo
         }
         else
         {
-            return hashOf(p[0 .. initializer().length]);
+            static import core.internal.hash;
+            const length = initializer().length;
+            if (length <= core.internal.hash.smallBytesHashThreshold)
+                return core.internal.hash.smallBytesHash((cast(const ubyte*)p)[0 .. length], 0);
+            else
+                return hashOf(p[0 .. initializer().length]);
```

Comment:
Can't `hashOf` handle this internally? They you don't need to expose two new symbols.

Reply:
It could. Updated the PR to do that.

Reply:
Oh. I thought that's what you meant, expose 1 symbol instead of 2.

Reply:
No, previously `hashOf` was called. Why not still calling that and handle everything inside that function?

Reply:
If I put that logic in `hashOf(ubyte[])` it would apply to more than just structs. I am cautious about that.